backend/benches/router.rs

```diff
@@ -39,7 +40,7 @@ fn benchmark_route(c: &mut Criterion) {
                 b.iter(|| {
                     let mut num_found = 0;
                     for (start, end, _) in &route_requests {
-                        if let Some(_found) = router.route_from_roads(*start, *end) {
+                        if let Some(_found) = router.route_from_roads(&router_input, *start, *end) {
```

Comment:
The danger is accidentally using the wrong `RouterInput` input for a `Router`. I'm not sure of a reasonable way to express "when the map model changes, the router gets invalidated" in Rust lifetimes. Does this seem OK to you?

Reply:
To make sure I'm understanding: 

The router graph's behavior is predicated on its input, so if its input changes, we need to rebuild the router. 

But that's an existing requirement, right? Or is there some new requirement related to that coming from #243?

Reply:
Your understanding is correct -- it is an existing requirement. But now for #243, we need to sometimes check what roads have a modal filter, and get its precise position. The new danger is calling `map.router_before(map.router_input_after(), pos1, pos2)` accidentally.

Reply:
Ah, I see! Yeah, that's a little rough... but I see why you came to this conclusion.

I think it's reasonable enough and we should go forward with it. Maybe we can revisit simplifying the code once we have #243 in place.